configs.go

```diff
@@ -105,3 +105,144 @@ func openAndReadFile(path string) ([]byte, error) {
 	}
 	return b, nil
 }
+
+// SessionPoolConf is the configs of a session pool
+// Note that the space name is bound to the session pool for its lifetime
+type SessionPoolConf struct {
+	username     string        // username for authentication
+	password     string        // password for authentication
+	serviceAddrs []HostAddress // service addresses for session pool
+	hostIndex    int           // index of the host in ServiceAddrs that the next new session will connect to
+	spaceName    string        // The space name that all sessions in the pool are bound to
+	sslConfig    *tls.Config   // Optional SSL config for the connection
+
+	// Basic pool configs
+	// Socket timeout and Socket connection timeout, unit: seconds
+	timeOut time.Duration
+	// The idleTime of the connection, unit: seconds
+	// If connection's idle time is longer than idleTime, it will be delete
+	// 0 value means the connection will not expire
+	idleTime time.Duration
+	// The max sessions in pool for all addresses
+	maxSize int
+	// The min sessions in pool for all addresses
+	minSize int
+}
+
+type SessionPoolConfOption func(*SessionPoolConf)
+
+// NewSessionPoolConfOpt creates a new NewSessionPoolConf with the provided options
+func NewSessionPoolConf(opts ...SessionPoolConfOption) (*SessionPoolConf, error) {
```

Comment:
if the user is mandatory, why not using  NewSessionPoolConf(user, password string, opts ...SessionPoolConfOption)